packages/react-components/react-utilities/src/compose/types.ts

```diff
@@ -33,16 +33,16 @@ export type UnknownSlotProps = Pick<React.HTMLAttributes<HTMLElement>, 'children
 /**
  * Helper type for {@link Slot}. Adds shorthand types that are assignable to the slot's `children`.
  */
-type WithSlotShorthandValue<Props extends { children?: unknown }> =
```

Comment:
by checking if `Props` extends `{ children?: unknown}` we're limiting the `Props` type to include in its signature `children` as a value, which is not valid for multiple cases (like on an `input` slot for example)